src/main/java/com/exactpro/th2/mstore/SessionBatchHolder.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.mstore;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+
+import org.jetbrains.annotations.NotNull;
+
+import com.exactpro.cradle.messages.StoredMessageBatch;
+import com.exactpro.cradle.utils.CradleStorageException;
+
+public class SessionBatchHolder {
+    private final AtomicLong lastSequence = new AtomicLong(Long.MIN_VALUE);
+
+    private final Supplier<StoredMessageBatch> batchSupplier;
+
+    private volatile Instant lastReset = Instant.now();
+
+    private volatile StoredMessageBatch holtBatch;
+
+    public SessionBatchHolder(Supplier<StoredMessageBatch> batchSupplier) {
+        this.batchSupplier = Objects.requireNonNull(batchSupplier, "'Batch supplier' parameter");
+        holtBatch = batchSupplier.get();
+    }
+
+    public long getAndUpdateSequence(long newLastSeq) {
+        return lastSequence.getAndAccumulate(newLastSeq, Math::max);
+    }
+
+    /**
+     * Tries to add the batch with messages to store to the currently holt batch.
+     * If batch can be added because of size or quantity limitations returns without changing the currently holt batch.
```

Comment:
If **a** batch **can't** be added because of **its** size or quantity limitations **, it will return the currently holt batch without any changing being performed**.

Reply:
The method does not return the batch itself. It completes without any changes being applied